compiler/qsc_circuit/Cargo.toml

```diff
@@ -19,6 +19,7 @@ qsc_data_structures = { path = "../qsc_data_structures" }
 qsc_frontend = { path = "../qsc_frontend" }
```

Comment:
I can't comment down below, so just putting this here. Looks like this Cargo.toml was missing the lint settings:
```
[lints]
workspace = true
```
And there are some warnings that need to be resolved. Search shows it's also missing from the qsc_formatter crate, but that deserves a separate issue.

Reply:
I've added this in and went and fixed the lints it found. Thanks for catching this!